apps/web/src/components/dashboard/energy-cost-card.tsx

```diff
@@ -32,9 +39,68 @@ export default async function EnergyCostCard({ startDate, endDate }: Props) {
+    const cost: number | null = price ? parseFloat((absolut * price).toFixed(2)) : null;
+
     const monthlyPayment = userData?.user_data.monthlyPayment;
 
+    const calculatePayment = () => {
+        if (monthlyPayment !== undefined && monthlyPayment !== null) {
+            const startYear = startDate.getFullYear();
+            const endYear = endDate.getFullYear();
+            const startMonth = startDate.getMonth();
+            const endMonth = endDate.getMonth();
+    
+            let totalAmount = 0;
+    
+            for (let year = startYear; year <= endYear; year++) {
+
+                const monthStart = (year === startYear) ? startMonth : 0;
+                const monthEnd = (year === endYear) ? endMonth : 11;
+    
+                for (let month = monthStart; month <= monthEnd; month++) {
+                    
+                    const firstDayOfMonth = (year === startYear && month === startMonth) ? startDate.getDate() : 1;
+                    const lastDayOfMonth = (year === endYear && month === endMonth) ? endDate.getDate() : new Date(year, month + 1, 0).getDate();
+                    const daysPerMonth = new Date(year, month + 1, 0).getDate();
+                    const paymentPerDay = monthlyPayment / daysPerMonth
+                    const daysInMonth = lastDayOfMonth - firstDayOfMonth + 1;
+                    const paymentPerMonth = paymentPerDay * daysInMonth
+                    totalAmount += paymentPerMonth;
+
+                }
+            }
+            return totalAmount.toFixed(2);
+        }
+    };
+
+    const calculateTotalConsumptionCurrentMonth = (data: EnergyEntry[]): number => {
+        const currentDate = new Date();
+        const currentMonthConsumptions = data.filter((entry: EnergyEntry) => {
+            const entryDate = entry.timestamp ? new Date(entry.timestamp) : null;
+            return entryDate && entryDate.getMonth() === currentDate.getMonth() && entryDate.getFullYear() === currentDate.getFullYear();
+        });
+        const totalConsumption = currentMonthConsumptions.reduce((total, entry) => total + entry.value, 0);
+      
+        return totalConsumption;
+      };
+
+
+    const calculatePredictedCost = (): number => {
+        const today: Date = new Date();
+        const firstDayOfMonth: Date = new Date(today.getFullYear(), today.getMonth(), 1);
+        const lastDayOfMonth: Date = new Date(today.getFullYear(), today.getMonth() + 1, 0);
+        const daysPassed: number = Math.floor((today.getTime() - firstDayOfMonth.getTime()) / (24 * 60 * 60 * 1000)) + 1;
+
+        const totalConsumptionCurrentMonth = calculateTotalConsumptionCurrentMonth(energyData);
+        const monthlyUsage: number = (totalConsumptionCurrentMonth / daysPassed) * lastDayOfMonth.getDate();
+        const daysRemaining: number = lastDayOfMonth.getDate() - daysPassed;
+
+        const predictedConsumption: number = ((monthlyUsage / daysPassed) * daysRemaining) / 1000;
```

Comment:
Wenn ich den Code richtig verstehe, willst du hier ja für predictedConsumption den Stromverbrauch ermitteln, den der Benutzer wahrscheinlich für den Rest des Monats haben wird, weil du ja später in Z. 101 cost + predictedCost rechnest (also die Kosten, die der Benutzer durch seinen bisherigen Verbrauch im Monat hatte, plus die Kosten, die der Benutzer wahrscheinlich für den Verbrauch im verbleibenden Monat haben wird. Wenn ich das falsch verstehe, korrigiere mich bitte :slightly_smiling_face: 

Wenn das stimmt: Kann man dann das dann nicht einfach so berechnen?
```suggestion
        const predictedConsumption: number = monthlyUsage - totalConsumptionCurrentMonth
```

Reply:
Jo du hast Recht. Hab gerade nochmal alles nachgerechnet. Die Variable monthlyUsage wurde ja auch schon extrapoliert auf den vollen Monat. Das wird dann für die Variable predictedConsumption doppelt durchgeführt und führt zu einem falschen Ergebnis. Danke für´s genaue hinschauen!🧐